### ISSUE TITLE
[BUG]: Executors aren't triggered when loading from a link

### ISSUE THREAD

**Issue by OfekShilon** [compiler-explorer/compiler-explorer#8959]
### Describe the bug

Must click 'recompile' on every executor to actually execute

### Steps to reproduce

-

### Expected behavior

-

### Reproduction link

https://compiler-explorer.com/z/fWc3nz1E5

### Screenshots

Not applicable

**Comment by adarshsm** [comment]
I think this comes down to an asymmetry between how the editor reacts to a compiler opening vs an executor opening on a short-link load.

When you load a `/z/...` link, the editor starts in `waitingForLanguage` mode (`static/panes/editor.ts`, `this.waitingForLanguage = Boolean(state.source && !state.lang)`) — it has source but no explicit language, and resolves the language only once a compiler/executor announces itself.

`onCompilerOpen` resolves the language before broadcasting:

```js
if (this.waitingForLanguage) {
    const glCompiler = _.find(
        this.container.layoutManager.root.getComponentsByName('compiler'),
        c => c.id === compilerId,
    );
    if (glCompiler) {
        const result = await this.hub.compilerService.processFromLangAndCompiler(null, glCompiler.originalCompilerId);
        if (result?.compiler) this.changeLanguage(result.compiler.lang);
    }
}
...
this.maybeEmitChange(true, compilerId);
```

`onExecutorOpen` does not:

```js
onExecutorOpen(executorId, editorId) {
    if (editorId === this.id) {
        this.maybeEmitChange(true);
        this.ourExecutors[executorId] = true;
    }
}
```

`maybeEmitChange` emits `editorChange` with `langId = this.currentLanguage?.id ?? ''`. Since `onExecutorOpen` never resolved the language, that `langId` is empty. The executor's `onEditorChange` then drops it because of the language guard (`static/panes/executor.ts`):

```js
if (editor === this.sourceEditorId && langId === this.currentLangId && compilerId === undefined) {
    this.source = source;
    if (this.settings.compileOnChange) this.compile();
}
```

So the source is never stored, `needsCompile` stays `false`, and `undefer()` has nothing to compile — which is why the executor sits idle until you click Recompile, while compilers on the same link run fine.

If that's right, the fix would be to give `onExecutorOpen` the same `waitingForLanguage` recovery as `onCompilerOpen` (resolve the language from the executor's compiler, `changeLanguage`, then `maybeEmitChange`). Happy to put up a PR along those lines — does that match what you'd expect?

**Comment by luantaraschi** [comment]
@adarshsm your diagnosis still holds on current `main` (58d1561e2). `onCompilerOpen` resolves the language behind `if (this.waitingForLanguage)` at `static/panes/editor.ts:1381`, and `onExecutorOpen` still goes straight to `maybeEmitChange(true)` with no equivalent.

One thing I ran into that makes it more than a copy of that block. The compiler path reaches the id through `glCompiler.originalCompilerId`, and `originalCompilerId` is not declared anywhere under `static/`: it only ever appears at that one read site, so it arrives on the GoldenLayout component from outside the typed surface. `getComponentsByName('executor')` has no counterpart, so the executor branch needs another way to reach the compiler id it was opened with, rather than the same lookup with the component name swapped.

Are you still planning to send this? If not I am happy to pick it up from your analysis, with the credit where it belongs. Just did not want to cut across work you had already done.
